### PR TITLE
overhaul of the xcm section

### PR DESCRIPTION
**Pull request by eshaben**
### Description

This PR includes an overhaul of the XCM section. I'll attempt to summarize all the changes 😅 

It adds the following:

- A core concepts section that includes information on XCM instructions, Multilocations, Weights and Fees
- A remote execution section that includes information on Computed Origins and guides on performing Remote Substrate Calls (split up by using the Substrate API and Ethereum API) and Remote EVM Calls
- A send XC-20s section that includes information on the XCM instructions used under the hood by the X-Tokens pallet, plus guides on sending XC-20s using the Substrate API and the Ethereum API
- A new way to better explain pallet and precompile interfaces, with information on parameters, returned data, and examples (where it makes sense)

It applies the following maintenance-related changes:

- Switches "multilocation-derivative" to Computed Origin
- Capitalizes Sovereign account
- Attempts to better explain weights and fees within the XCM Transactor, X-Tokens, etc. pages
- Reorder images and snippets to mirror the new directory structure
- Reverts some of the links in the weights and fees page to the old substrate repository, as Kusama and Polkadot are still on spec version v0.9.43 so we should refer to those pages even though the repo has been archived. The links will, of course, be updated for the next version (which should then point to the new repo)

### Checklist

- [x] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [x] If pages have been moved around, I have run the `move-pages.py` script to move the pages and update the image paths on the chinese repo
    - [x] After the script has been run, I have created an additional PR in `moonbeam-docs-cn`
- [x] If images have been added, I have run the `compress-images.py` script to compress the images.
- [x] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables

### Corresponding PRs

~~Expandable function documentation: https://github.com/papermoonio/moonbeam-mkdocs/pull/151~~

Redirects: https://github.com/papermoonio/moonbeam-mkdocs/pull/155
CN path updates: https://github.com/moonbeam-foundation/moonbeam-docs-cn/pull/362

### After Translation Requirements

- [x] Will need to create PR in `moonbeam-docs` repo to remove images
- [x] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site

#### Items to be Updated

Redirects: The same redirects as applied to the `mkdocs.yml `file need to be applied to the `mkdocs-cn/mkdocs.yml` (see PR in Corresponding PRs section)
Images: Delete `builders/interoperability/xcm/fees/*` images

### CODE REVIEW

**Comment by eshaben** [comment]
To render the new documentation feature (expandable documentation on parameters, returned data, examples) in this PR correctly, you'll need to switch your mkdocs branch to `eshaben/function-admonition`. Associated PR: https://github.com/papermoonio/moonbeam-mkdocs/pull/151

File: builders/interoperability/xcm/remote-execution/.pages
Comment (themacexpert) [inline builders/interoperability/xcm/remote-execution/.pages:1]: Screenshot 2023-10-31 at 7 01 36 PM
One of the descriptions is missing a period at the end.

**Review by themacexpert — APPROVED** [review]
LGTM 🚢

**Comment by eshaben** [comment]
@albertov19 this is finally ready! 🚀

**Pull request merged**